src/probnum/filtsmooth/gaussfiltsmooth/unscentedkalman.py

```diff
@@ -77,11 +77,13 @@ def __init__(self, dynamod, measmod, initrv, alpha, beta, kappa, **kwargs):
         else:
             self.cke_nsteps = 1
         super().__init__(dynamod, measmod, initrv)
-        self.ut = UnscentedTransform(self.dynamod.ndim, alpha, beta, kappa)
+        self.ut = UnscentedTransform(self.dynamod.dimension, alpha, beta, kappa)
```

Comment:
I made the change, not because I think they are too short, but rather because `ndim` is a property of arrays (and as such, properties of `RandomVariable` and `LinearOperator` and so on). It is a bit out of nowhere, I admit that.

I am not entirely sure whether there is a possibility to mix them up accidentally, but when I did this I thought it would be useful to use a different name for something like a spatial dimension as compared to the number of dimensions of a matrix.
What do you think?

Reply:
Ok, I understand the issue. It's true that in that ocntext `ndim` would be misleading, especially since it refers to a totally different thing than what we call `dimension` now. 

The only alternatives I could think of would be `size` or `len`/`length`, since numpy uses those when refering to the number of elements of arrays (and the point of view would be that we mean the size of the "state" vector, not the transition dynamics).

I'm also fine with keeping "dimension" :)

Reply:
Might be too opinionated, but I think it is best to steer clear of all numpy-associated terms, including `size` and `len`. I, personally, like `dimension`, but in the ODEPrior we use sth like `spatialdim` which I also find somewhat explanatory. 
In any case, I think it makes sense to pass through the ODE/Filter code to check for remaining `ndim` (i.e. what do we use in `IVP`?).

Reply:
Both `dimension` and `spatialdim` would be fine for me. Does the `spatialdim` describe exactly the same thing that is now called `dimension`? If yes we should use the same name for both, otherwise feel free to keep it different.

Good idea also to pass through once more to check for `ndim`s! :)